gpytorch/variational/variational_distribution.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import torch
+from ..lazy import CholLazyTensor
+from ..distributions import MultivariateNormal
+from ..module import Module
+
+
+class VariationalDistribution(Module):
+    """
+    VariationalDistribution objects represent the variational distribution q(u) over a set of inducing points for GPs.
+
+    The most common way this distribution is defined is to parameterize it in terms of a mean vector and a covariance
+    matrix. In order to ensure that the covariance matrix remains positive definite, we only consider the lower triangle
+    and we manually ensure that the diagonal remains positive.
+    """
+
+    def __init__(self, num_inducing_points, batch_size=None):
+        """
+        Args:
+            num_inducing_points (int): Size of the variational distribution. This implies that the variational mean
+                should be this size, and the variational covariance matrix should have this many rows and columns.
+            batch_size (int, optional): Specifies an optional batch size for the variational parameters. This is useful
+                for example when doing additive variational inference.
+        """
+        super(VariationalDistribution, self).__init__()
+        mean_init = torch.zeros(num_inducing_points)
+        covar_init = torch.eye(num_inducing_points, num_inducing_points)
+        if batch_size is not None:
+            mean_init = mean_init.repeat(batch_size, 1)
+            covar_init = covar_init.repeat(batch_size, 1, 1)
+        mean_init += torch.randn_like(mean_init).mul(1e-1)
+        covar_init += torch.randn_like(covar_init).mul(1e-1)
+
+        self.register_parameter(name="variational_mean", parameter=torch.nn.Parameter(mean_init))
+        self.register_parameter(name="chol_variational_covar", parameter=torch.nn.Parameter(covar_init))
+
+        self.register_buffer("variational_params_initialized", torch.tensor(0))
+
+    @property
+    def variational_distribution(self):

```

Comment:
Suuuuper minor point, but could this be a 'CholeskyVariationalStrategy' - which uses the super class 'VariationalStrategy' which has an abstract method for forward?